libpromises/dbm_lmdb.c

```diff
@@ -558,7 +559,28 @@ DBPriv *DBPrivOpenDB(const char *const dbpath, const dbid id)
     open_flags |= MDB_WRITEMAP;
 #endif
 
-    rc = LmdbEnvOpen(db->env, dbpath, open_flags, 0644);
+    // If effective user is root then change to system group for lmdb files
+    // otherwise leave group as-is.
+    uid_t p_euid = geteuid();
+    gid_t current_gid = getgid();
+    if (p_euid == 0)
+    {
+        gid_t system_gid = Str2Gid(CF_SYSTEM_GROUP, NULL /* no groupcopy */, NULL /* no Promise */);
+        if (system_gid == CF_SAME_GROUP || system_gid == CF_UNKNOWN_GROUP)
+        {
+            Log(LOG_LEVEL_ERR, "Could not get gid_t for CF_SYSTEM_GROUP('%s'), got %d", CF_SYSTEM_GROUP, system_gid);
```

Comment:
Only thing I don't like about this part of the code is that if CF_SYSTEM_GROUP doesn't exist (not likely right? see above definitions.h) then we will get this error many times, once for each LMDB file processed/created.

If we decide to change group to system for the entire agent run, obviously this multiple errors wouldn't be a problem as we would only change once.